tests/test_item.py

```diff
@@ -50,3 +52,86 @@ def test_read_eo_item_owns_asset(self):
+
+
+class CommonMetadataTest(unittest.TestCase):
+    def setUp(self):
+        self.URI_1 = TestCases.get_path(
+            'data-files/examples/0.9.0/item-spec/examples/datetimerange.json')
+        self.ITEM_1 = Item.from_file(self.URI_1)
+
+        self.URI_2 = TestCases.get_path(
+            'data-files/examples/0.9.0/item-spec/examples/sample-full.json')
+        self.ITEM_2 = Item.from_file(self.URI_2)
+
+        self.EXAMPLE_CM_DICT = {
+            'start_datetime':
+            '2020-05-21T16:42:24.896Z',
+            'platform':
+            'example platform',
+            'providers': [{
+                'name': 'example provider',
+                'roles': ['example roll'],
+                'url': 'https://example-provider.com/'
+            }]
+        }
+
+    def test_datetimes(self):
+        # save dict of original item to check that `common_metadata`
+        # method doesn't mutate self.item_1
+        before = self.ITEM_1.clone().to_dict()
+        start_datetime_str = self.ITEM_1.properties['start_datetime']
+        self.assertIsInstance(start_datetime_str, str)
+
+        common_metadata = self.ITEM_1.common_metadata
+        self.assertIsInstance(common_metadata, CommonMetadata)
+        self.assertIsInstance(common_metadata.start_datetime, datetime)
+        self.assertDictEqual(before, self.ITEM_1.to_dict())
+        self.assertIsNone(common_metadata.providers)
+
+    def test_set_common_metadata(self):
+        common_metadata = self.ITEM_2.common_metadata
+
+        for provider in common_metadata.providers:
+            self.assertIsInstance(provider, Provider)
+        self.assertEqual(common_metadata.platform, 'coolsat2')
+
+        example_cm = CommonMetadata(self.EXAMPLE_CM_DICT)
+        self.assertIsInstance(example_cm.start_datetime, datetime)
+        self.assertIsInstance(example_cm.providers[0], Provider)
+
+        item_2_copy = self.ITEM_2.clone()
+
+        item_2_copy.set_common_metadata(example_cm)
+        self.assertNotEqual(self.EXAMPLE_CM_DICT['platform'], item_2_copy.properties['platform'])
+        self.assertEqual(item_2_copy.properties['platform'], self.ITEM_2.properties['platform'])
+        item_2_copy.set_common_metadata(example_cm, override=True)
+
+        self.assertEqual(self.EXAMPLE_CM_DICT['platform'], item_2_copy.properties['platform'])
+        self.assertNotEqual(item_2_copy.properties['platform'], self.ITEM_2.properties['platform'])
+        self.assertIsInstance(item_2_copy.properties['start_datetime'], str)
+        self.assertEqual(item_2_copy.properties['start_datetime'],
+                         self.EXAMPLE_CM_DICT['start_datetime'])
+        self.assertEqual(item_2_copy.properties['platform'], self.EXAMPLE_CM_DICT['platform'])
+        for i, provider in enumerate(item_2_copy.properties['providers']):
+            self.assertDictEqual(provider, self.EXAMPLE_CM_DICT['providers'][i])
+
+    def test_common_metadata_getter_setters(self):
```

Comment:
Does this cover all of the property getter/setters that have non-trival logic?

Not necessary but consider breaking each property that has non-trivial logic into it's own test for clarity.